src/modules/hip/kernel/flip.hpp

```diff
@@ -0,0 +1,311 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+
+template <typename T>
+__global__ void flip_pkd_tensor(T *srcPtr,
+                                uint2 srcStridesNH,
+                                T *dstPtr,
+                                uint2 dstStridesNH,
+                                unsigned int *horizontalTensor,
+                                unsigned int *verticalTensor,
+                                RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    d_float24 pix_f24;
+    uint horizontalFlag, verticalFlag, srcIdx = id_z * srcStridesNH.x;
+    horizontalFlag = horizontalTensor[id_z];
+    verticalFlag = verticalTensor[id_z];
+
+    if(horizontalFlag == 0 && verticalFlag == 0)
+    {
+        srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3;
+        rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &pix_f24);
+    }
+    else if(horizontalFlag == 1 && verticalFlag == 1)
+    {
+        srcIdx = (id_z * srcStridesNH.x) + ((roiTensorPtrSrc[id_z].xywhROI.xy.y + roiTensorPtrSrc[id_z].xywhROI.roiHeight - 1 - id_y) * srcStridesNH.y) + (roiTensorPtrSrc[id_z].xywhROI.xy.x + roiTensorPtrSrc[id_z].xywhROI.roiWidth - id_x - 8) * 3;
```

Comment:
Do you see a benefit of using roi in ltrb instead of xywh? If yes, you can use hip_exec_roi_converison_xywh_to_ltrb to convert it before kernel launch.